src/precompile_utils.c

```diff
@@ -189,7 +189,7 @@ static int precompile_enq_specialization_(jl_method_instance_t *mi, void *closur
             jl_value_t *inferred = jl_atomic_load_relaxed(&codeinst->inferred);
             if (inferred &&
                 inferred != jl_nothing &&
-                (jl_ir_inlining_cost(inferred) == UINT16_MAX)) {
+                (jl_options.compile_enabled != JL_OPTIONS_COMPILE_ALL && jl_ir_inlining_cost(inferred) == UINT16_MAX)) {
```

Comment:
did you mean to write `jl_options.compile_enabled == JL_OPTIONS_COMPILE_ALL || ` here instead?

Reply:
Oooooh, good catch. Yes, I do believe that's what we intended.  This was meant to be the "softer" #53547, where we only considered the inlineability heuristics when we're not compile_all'ing... intending to otherwise compile even small functions.